File: 0055-jump-game/0055-jump-game.go
func canJump(nums []int) bool {
    var maxPosition int = 0
     var length int = len(nums) - 1

    
    for i:= 0; i <= length; i++ {
        if maxPosition >= length {
            return true
        }
        if nums[i] == 0 && maxPosition == i {
            return false;
        }
        if nums[i] + i > maxPosition {
            maxPosition = nums[i] + i
        }
    }
    
    return true
}

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxPosition = 0;
        
        for(int i = 0; i < nums.length; i++){
            if(maxPosition >= nums.length - 1) return true;
            if(nums[i] == 0 && maxPosition == i) return false;
            if(nums[i] + i > maxPosition) maxPosition = nums[i] + i ;
        }
        
        return true;
    }
}